backend/src/GraphQl/AccessRights/DataAccessRightsMutations.cs

```diff
@@ -0,0 +1,212 @@
+        if (data == null)
+        {
+            return new AddDataAccessRightsPayload(
+                new AddDataAccessRightsError(
+                    AddDataAccessRightsErrorCode.UNKNOWN_DATA,
+                    $"Unknown data.",
+                    []
+                )
+            );
+        }
+
+        data.DataAccess = input.DataAccessMode;
+        if (data.DataAccess == Enumerations.DataAccessMode.RESTRICTED)
+        {
+            data.DataAccessRights.AllowedInstitutions = input.DataAccessRights.AllowedInstitutions;
+
+            data.DataAccessRights.AllowedApplications = input.DataAccessRights.AllowedApplications;
+
+            data.DataAccessRights.AllowedUserAndQuantity = input.DataAccessRights.AllowedUserAndQuantity;
+        }
+
+        await context.SaveChangesAsync(cancellationToken).ConfigureAwait(false);
+        return new AddDataAccessRightsPayload(data.DataAccessRights);
+    }
+
+    public async Task<AddAccessRightsPayload> AddAccessRightsAsync(
+        AccessRightsInput input,
+        ApplicationDbContext context,
+        AppSettings appSettings,
+        IHttpClientFactory httpClientFactory,
+        IHttpContextAccessor httpContextAccessor,
+        IUserService userService,
+        IDataService dataService,
+        CancellationToken cancellationToken
+    )
+    {
+        var currentUser = await userService.GetCurrentUser(
+            cancellationToken).ConfigureAwait(false);
+        if (currentUser == null)
+        {
+            return new AddAccessRightsPayload(
+                new AddAccessRightsError(
+                    AddAccessRightsErrorCode.UNAUTHENTICATED,
+                    $"The user is not authenticated.",
+                    []
+                )
+            );
+        }
+
+        //if (!CommonAuthorization.(
+        //    currentUser,
+        //    input.CreatorId,
+        //    appSettings,
+        //    httpClientFactory,
+        //    httpContextAccessor,
+        //    cancellationToken
+        //    )
+        //)
+        //    return new AddDataAccessRightsPayload(
+        //        new AddDataAccessRightsError(
+        //            AddDataAccessRightsErrorCode.UNAUTHORIZED,
+        //            $"The current user is not authorized to approval for the institution.",
+        //            []
+        //        )
+        //    );
+
+        var accessRights = await context.AccessRights.FirstOrDefaultAsync(x => x.InstitutionId == input.InstitutionId, cancellationToken).ConfigureAwait(false);
+
+        if (accessRights is not null)
+        {
+            return new AddAccessRightsPayload(
+                new AddAccessRightsError(
+                    AddAccessRightsErrorCode.ALREADY_EXISTS,
+                    $"The access rights for this institution already exist.",
+                    []
+                )
+            );
+        }
+        else
```

Comment:
Remove this `else`. Because the `if` above returns, this `else` is not necessary.